mypy/checker.py

```diff
@@ -179,11 +179,25 @@ def check_first_pass(self) -> None:
 
         all_ = self.globals.get('__all__')
         if all_ is not None and all_.type is not None:
-            seq_str = self.named_generic_type('typing.Sequence',
-                                              [self.named_type('builtins.str')])
-            if not is_subtype(all_.type, seq_str):
-                str_seq_s, all_s = self.msg.format_distinctly(seq_str, all_.type)
-                self.fail(messages.ALL_MUST_BE_SEQ_STR.format(str_seq_s, all_s),
+            valid_all_types = [self.named_generic_type('typing.Sequence',
+                                                       [self.named_type('builtins.str')])]
+            if self.options.python_version[0] < 3:
+                valid_all_types.append(
+                    self.named_generic_type('typing.Sequence',
```

Comment:
Since mypy considers `str` compatible with `unicode` and `Sequence` is covariant, only checking against `Sequence[unicode]` should be sufficient in Python 2 mode, instead of both `Sequence[str]` and `Sequence[unicode]`. So the only needed change would be making `seq_str` depend on Python major version.